core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java

```diff
@@ -213,6 +216,8 @@
   protected NewBookmarksDao newBookmarksDao;
   @Inject
   protected NewBookDao newBookDao;
+  @Inject
+  private AlertDialogShower alertDialogShower;
```

Comment:
```
e: /home/runner/work/kiwix-android/kiwix-android/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java:220: error: Dagger does not support injection into private fields
107
  private AlertDialogShower alertDialogShower;
```

You can't inject into private fields so dagger has failed the build. This means you were bypassing the githooks on commit and additionally that you did not run this code on a device or emulator. These are 2 crucial steps in development.

Reply:
Thanks! I was going to ask about this after you review. I always build for the actual PR, I just wanted some feedback for the edits.
For some reason though, I did not get that build failure message I got this one
```
D:\Projects\kiwix\android\kiwix-android\core\src\main\java\org\kiwix\kiwixmobile\core\data\local\dao\BookDao.java:43: error: cannot find symbol
  private void setBookDetails(Book book, SquidCursor<BookDatabaseEntity> bookCursor) {
                                                     ^
  symbol:   class BookDatabaseEntity
  location: class BookDao
```

Reply:
There should be many errors, try toggling between the views in the console
![image](https://user-images.githubusercontent.com/3358282/75164676-0e10c380-5719-11ea-96a6-6a4082efb0a2.png)
That 2 squares button on the left hand side